applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_small_strain_u_pw_element.cpp

```diff
@@ -0,0 +1,545 @@
+{
+    // Arrange
+    const auto p_geometry =
+        std::make_shared<Triangle2D3<Node>>(ModelSetupUtilities::Create2D3NTriangleGeometry());
+    const auto                        p_properties = std::make_shared<Properties>();
+    const UPwSmallStrainElement<2, 3> element(0, p_geometry, p_properties,
+                                              std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_DoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    Model model;
+    const auto p_element = CreateUPwSmallStrainElementWithUPwDofs(model, std::make_shared<Properties>());
+
+    // Act
+    Element::DofsVectorType degrees_of_freedom;
+    p_element->GetDofList(degrees_of_freedom, ProcessInfo{});
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 9);
+    std::vector<std::string> variable_names;
+    std::transform(degrees_of_freedom.cbegin(), degrees_of_freedom.cend(), std::back_inserter(variable_names),
+                   [](const auto& rpDof) { return rpDof->GetVariable().Name(); });
+    const std::vector<std::string> desired_variable_list{
+        "DISPLACEMENT_X", "DISPLACEMENT_Y", "DISPLACEMENT_X", "DISPLACEMENT_Y", "DISPLACEMENT_X",
+        "DISPLACEMENT_Y", "WATER_PRESSURE", "WATER_PRESSURE", "WATER_PRESSURE"};
+
+    KRATOS_EXPECT_EQ(variable_names, desired_variable_list);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_IntegrationMethod, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_geometry =
+        std::make_shared<Triangle2D3<Node>>(ModelSetupUtilities::Create2D3NTriangleGeometry());
+    const auto                        p_properties = std::make_shared<Properties>();
+    const UPwSmallStrainElement<2, 3> element(0, p_geometry, p_properties,
+                                              std::make_unique<PlaneStrainStressState>());
+
+    // Act
+    const auto p_integration_method = element.GetIntegrationMethod();
+
+    // Assert
+    constexpr auto expected_integration_method = GeometryData::IntegrationMethod::GI_GAUSS_2;
+    KRATOS_EXPECT_EQ(p_integration_method, expected_integration_method);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UPwSmallStrainElement_CheckDoesNotThrowOnCorrectInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    Model model;
+    auto  element = CreateUPwSmallStrainElementWithUPwDofs(model, CreateProperties());
+    SetSolutionStepValuesForGeneralCheck(element);
+    auto process_info = ProcessInfo{};
+
+    // Act, no exceptions on correct input
+    KRATOS_EXPECT_EQ(element->Check(process_info), 0);
```

Comment:
No need to use an intermediate variable here (the class name is just as clear as the variable name):
```suggestion
    // Act, no exceptions on correct input
    KRATOS_EXPECT_EQ(element->Check(ProcessInfo{}), 0);
```

Reply:
done